### PR TITLE
Update ApplicationDbContextFactory

### PR DESCRIPTION
**Pull request by bricelam**
Depends on aspnet/EntityFramework#8314

@Eilon @DamianEdwards @ajcvickers

### CODE REVIEW

**Comment by bricelam** [comment]
@Eilon So nice that I'm starting to second guess if we even need to do this automatically...

Thoughts, @DamianEdwards?

**Comment by Eilon** [comment]
Hmmm @DamianEdwards ?

Why wouldn't we want this?

**Comment by prafullbhosale** [comment]
:eyes:

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Company.WebApplication1.Data
 {
     public class ApplicationDbContextFactory : IDbContextFactory<ApplicationDbContext>
     {
-        public ApplicationDbContext Create(DbContextFactoryOptions options) =>
-            Program.BuildWebHost(Array.Empty<string>()).Services.GetRequiredService<ApplicationDbContext>();
+        public ApplicationDbContext Create(string[] args) =>
+            Program.BuildWebHost(args).Services.GetRequiredService<ApplicationDbContext>();
```
Comment (bricelam) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: May change our signature to `IEnumerable<string>`. Should I call `.ToArray()` here or change the signature of `BuildWebHost` too?

Comment (bricelam) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Given that both [WebHost.CreateDefaultBuilder](https://github.com/aspnet/MetaPackages/blob/rel/2.0.0-preview1/src/Microsoft.AspNetCore/WebHost.cs#L150) and [IConfigurationBuilder.AddCommandLine](https://github.com/aspnet/Configuration/blob/rel/2.0.0-preview1/src/Microsoft.Extensions.Configuration.CommandLine/CommandLineConfigurationExtensions.cs#L21) (which it calls) are `string[]`, we should keep it `string[]`.

**Comment by bricelam** [comment]
@mlorbetske upstream PRs have been merged. This one is good to go.

**Comment by mlorbetske** [comment]
@bricelam These require new packages, right?

**Comment by danroth27** [comment]
Yup, the latest release build has these changes

**Pull request merged**